GeeksCoreLibrary/Components/OrderProcess/OrderProcess.cs

```diff
@@ -434,7 +435,9 @@ private async Task<string> HandleCheckoutModeAsync()
                         break;
                     case OrderProcessStepTypes.OrderConfirmation:
                     case OrderProcessStepTypes.OrderPending:
-                        var confirmationHtml = ReplaceEntityDataInTemplate(shoppingBasket, currentItems, step, steps, paymentMethods);
+                        var orderId = Convert.ToUInt64(httpContextAccessor.HttpContext.Request.Query["order"].ToString().Decrypt());
```

Comment:
Kun je hier wat controles inbouwen? Wat als er geen order ID in de URL staat? Dan krijg je waarschijnlijk een NullReferenceException. En wat als daar een waarde in staat die niet gedecrypt kan worden? Dan krijg je een 500 terug. In zo'n gevallen kun je beter wel gewoon de bedanktpagina tonen, maar dan zonder orderinformatie, of met een vriendelijke foutmelding.

Reply:
Extra validatie is toegevoegd om dit te voorkomen. Alle 4 de flows (succes en 3x fail) zijn getest.